Lib/test/test_functools.py

```diff
@@ -222,6 +222,26 @@ def test_repr(self):
                       [f'{name}({capture!r}, {args_repr}, {kwargs_repr})'
                        for kwargs_repr in kwargs_reprs])
 
+    def test_equality(self):
+        p = functools.partial(capture, 1, 2, a=10, b=20)
+        q = functools.partial(capture, 1, 2, a=10, b=20)
+        self.assertTrue(p == q)
+        self.assertFalse(p != q)
+        self.assertTrue(p.__eq__(q))
+        self.assertFalse(p.__ne__(q))
+
+        q = self.partial(capture, 1, 2, a=10)
+        self.assertFalse(p == q)
+        self.assertTrue(p != q)
+
+        self.assertNotEqual(p, capture)
+        self.assertNotEqual(q, capture)
+
+        a = self.partial(capture)
+        b = self.partial(signature)
+        self.assertFalse(a == b)
+        self.assertTrue(a != b)
```

Comment:
You should also check comparisons against other types, and check comparisons with bound arguments that raise exceptions when compared.